# Project2_firstdraft
This is a first draft upload of Project 2.

The following files are needed to run the code in R studio:
1) Human and Mouse PPAR alpha fasta files (uploaded here)
2) Human PPAR alpha ligand binding domain crystal structure (uploaded here)

The Human PPAR alpha and Mouse PPARA alpha protein sequence files were obtained from Uniprot with the accession IDs Q07869 and P23204 respectively (https://beta.uniprot.org/uniprotkb/Q07869/entry) (https://beta.uniprot.org/uniprotkb/P23204/entry). The sequences saved as fasta files were used to perform pairwise sequence alignment to compare the protein sequences of human PPAR alpha and mouse PPAR alpha.
The Human PPAR alpha ligand binding domain crystal structure data was obtained from the Protein Data Bank with the accession ID 2ZNN (https://www.rcsb.org/structure/2ZNN). The structure was used as a template to make a homology model using the mouse PPAR alpha protein sequence, since a mouse PPAR alpha crystal structure is not available.

The scientific question and hypothesis are as follows:
Scientific Question: Do the amino acid sequences and structure between the peroxisome proliferator-activated receptor alpha in humans and mice share enough similarity to consider the PPAR alpha mice model an accurate model of the PPAR alpha receptor in humans?

Hypothesis: If the peroxisome proliferator-activated receptor alpha is mainly activated through the ligand binding domain, then there should be at least 85% similarity between the ligand binding domain when comparing the protein sequences and structures of PPAR alpha between human and mouse.
